cheroot/connections.py

```diff
@@ -61,9 +62,11 @@ def __init__(self, server):
                 that uses this ConnectionManager instance.
         """
         self.server = server
-        self.connections = []
+        self._readable_conns = collections.deque()
```

Comment:
general restructuring: only maintain connections that the selector has previously indicated are ready to read in `_readable_conns`.

all other connections remain registered with the selector, to avoid registering/unregistering them unnecessarily.